### DISCUSSION TITLE
[Warning] Access denied for user

### DISCUSSION THREAD

**Discussion by kwenrick1** [deeztek/Hermes-Secure-Email-Gateway#35] [Support] [answered]
It looks like after the script finished I am missing users in mariadb. How do I fix this? 

Ubuntu 20.04 LTS

Error on webpage login
Lucee 5.2.9.31 Error (database)
Access denied for user ''@'localhost' (using password: YES)

/var/log/mysql/error.log
2022-10-17 22:15:01 448 [Warning] Access denied for user ''@'localhost' (using password: YES)
2022-10-17 22:15:04 449 [Warning] Access denied for user 'hermes'@'localhost' (using password: YES)
2022-10-17 22:17:01 450 [Warning] Access denied for user 'syslog'@'localhost' (using password: YES)

MariaDB [(none)]> SELECT User, Db, Host from mysql.db;
+------+-----------+-----------+
| User | Db        | Host      |
+------+-----------+-----------+
|      | Syslog    | 127.0.0.1 |
|      | djigzo    | 127.0.0.1 |
|      | hermes    | 127.0.0.1 |
|      | opendmarc | 127.0.0.1 |
|      | Syslog    | localhost |
|      | djigzo    | localhost |
|      | hermes    | localhost |
|      | opendmarc | localhost |
+------+-----------+-----------+
8 rows in set (0.000 sec)

MariaDB [(none)]> SELECT User FROM mysql.user;
+------+
| User |
+------+
|      |
|      |
| root |
+------+
3 rows in set (0.000 sec)

**Comment by deeztek** [comment]
I'm guessing you got some errors during install. Can you provide the install script log located under the same directory you started the installation script?

Thanks

**Comment by kwenrick1** [comment]
[install_log-10-14-2022-1601.log](https://github.com/deeztek/Hermes-Secure-Email-Gateway/files/9832235/install_log-10-14-2022-1601.log)

**Comment by deeztek** [comment]
There is nothing in that log file besides the following entry:

`[10/14/2022-16:01] Starting Hermes SEG 18.04 Installation.`

This looks like something went very wrong with the installation. Did anything get installed? You can also jump in Matrix if you wish to have more real time discussion on the issue you are experiencing:

[https://matrix.to/#/#hermesseg:matrix.org](https://matrix.to/#/#hermesseg:matrix.org)

Thanks

**Comment by kwenrick1** [comment]
[install_log-10-14-2022-1604.log](https://github.com/deeztek/Hermes-Secure-Email-Gateway/files/9832566/install_log-10-14-2022-1604.log)

Unfortunately I've run through the installer a few times trying to get things working. That is the earliest I have that has data in it

**Reply by deeztek** [reply]
ok this one has more data in it. So this particular file says that you did not supply a MySQL root password: 

`Access denied for user 'root'@'localhost' (using password: NO)`

When you start the installation, the script asks you to create passwords for the MySQL root as well as usernames and passwords for the hermes, Syslog, opendmar and djigzo databases. Did you supply those credentials you wish to use? Because it looks like you did not (Note the user is blank in the lines below):

```
[10/14/2022-16:07] Creating Hermes SEG database and assigning user  to it
ERROR 1007 (HY000) at line 1: Can't create database 'hermes'; database exists
[10/14/2022-16:07] Creating Syslog database and assigning user  to it
ERROR 1007 (HY000) at line 1: Can't create database 'Syslog'; database exists
[10/14/2022-16:07] Creating Opendmarc database and assigning user  to it
ERROR 1007 (HY000) at line 1: Can't create database 'opendmarc'; database exists
[10/14/2022-16:07] Creating Ciphermail database and assigning user  to it
ERROR 1007 (HY000) at line 1: Can't create database 'djigzo'; database exists

```
 If so, did the passwords you chose contain any special characters? If so, what special characters were they? 

Thanks

**Comment by kwenrick1** [comment]
I supplied them. I believe the log file is from the 2nd completion of the
install attempt. I no longer have the first install log file

**Reply by deeztek** [reply]
Okay in that case, I suggest you remove mariadb:

`apt-get purge mariadb-server mariadb-client -y`

Then remove the following directory:

`rm -rf /mnt/data/dbase`

Start the installation again, ensuring you supply a MySQL root password as welll as usernames and passwords for the hermes, syslog, djigzo and opendmarc databases. Make sure you do not supply any special characters for the passwords to play it safe. Besides in security password length always beats password complexity. So, if you want to be safe create a 30 character alphanumeric password.

If it fails, provide the install_log file again.

**Comment by kwenrick1** [comment]
Install ended in some errors. 

[install_log-10-20-2022-1443.log](https://github.com/deeztek/Hermes-Secure-Email-Gateway/files/9833271/install_log-10-20-2022-1443.log)


Errors on the webpage after login:

Lucee 5.2.9.31 Error (database)
--
Message | Access denied for user ''@'localhost' (using password: YES)
Stacktrace | The Error Occurred in/var/www/html/admin/2/inc/get_console_mode.cfm: line 2422: <cfquery name = "getconsolemode" datasource="hermes">23: select parameter, value2 from parameters2 where parameter='console.mode'24: </cfquery>25:26: <cfset theConsoleMode="#getconsolemode.value2#">called from /var/www/html/admin/Application.cfc: line 84

**Reply by deeztek** [reply]
Looking at the new install_log-10-20-2022-1443.log file you sent, I'm still noticing the following errors:

```
[10/20/2022-14:44] Starting MySQL(MariaDB) Database
[10/20/2022-14:44] Creating Hermes SEG database and assigning user  to it
[10/20/2022-14:44] Creating Syslog database and assigning user  to it
[10/20/2022-14:44] Creating Opendmarc database and assigning user  to it
[10/20/2022-14:44] Creating Ciphermail database and assigning user  to it
[10/20/2022-14:44] Creating Hermes SEG Database Schema
[10/20/2022-14:45] Creating Opendmarc database schema.
```

For instance the line:

`Creating Hermes SEG database and assigning user  to it`

is supposed to say:

`Creating Hermes SEG database and assigning user [someuser] to it`

where **[someuser]** is the username you entered when the script prompted you. 

So, the question is, are you removing the actual usernames from the install_log before you send it in order to sanitize it or are you not entering the username when prompted?

This is what it's supposed to look like where **hermes, rsyslog, opendmarc** and **ciphermail** are the usernames I entered when the script prompted me:

```
[08/01/2022-12:33] Creating Hermes SEG database and assigning user hermes to it
[08/01/2022-12:33] Creating Syslog database and assigning user rsyslog to it
[08/01/2022-12:33] Creating Opendmarc database and assigning user opendmarc to it
[08/01/2022-12:33] Creating Ciphermail database and assigning user ciphermail to it
[08/01/2022-12:33] Creating Hermes SEG Database Schema
[08/01/2022-12:33] Creating Opendmarc database schema.
```

**Comment by kwenrick1** [comment]
log file is unedited. During install I would be prompted for something like
this
Enter MySQL(MariaDB) Hermes SEG user username you wish to use (Example:
hermes):

I would hit enter. Does a username need entered or does it default to the
suggested?

**Reply by deeztek** [reply]
Ah I see. Those are not suggested/default values. Those are examples. You must fill in all the information you are prompted for. We will make them mandatory next update release.

Thanks and sorry for the confusion.

**Comment by deeztek** [comment, marked as answer]
Added new enhancement #36 for next release.

**Comment by kwenrick1** [comment]
Actually typing usernames did the trick. Thanks for pointing that out

**Reply by deeztek** [reply]
np glad I could help.
